### ISSUE TITLE
Operator "line" does not work with "--input-format json".

### ISSUE THREAD

**Issue by Kopfbremse** [mikefarah/yq#1422]
**Version of yq**: 4.29.2
**Operating system**: linux
**Installed via**: binary release

**Describe the bug**
the `line` operator does not work for `--input-format json`

**Input JSON**
```json
{
  "a": 1,
  "b": {
    "c": 2,
    "d": {
      "e": 3
    }
  }
}
```

**Command**
```bash
yq e '[.. | {"line": line, "path": path | join("."), "type": type}]'
```

**Output**
```yaml
- line: 0
  path: ""
  type: ""
- line: 1
  path: a
  type: '!!int'
- line: 0
  path: b
  type: ""
- line: 1
  path: b.c
  type: '!!int'
- line: 0
  path: b.d
  type: ""
- line: 1
  path: b.d.e
  type: '!!int'
```

**Expected behaviour**
It would be nice to get the correct line of the nodes.
If this is not possible, as with YAML, the output should always be `line: 0`.

**Comment by mikefarah** [comment]
Fixed in 4.30.5 (line is 0)

**Comment by Kopfbremse** [comment]
thank you
